src/monodroid/jni/strings.hh

```diff
@@ -0,0 +1,731 @@
+#ifndef __STRINGS_HH
+#define __STRINGS_HH
+
+#include <array>
+#include <cstring>
+#include <cerrno>
+#include <limits>
+#include <type_traits>
+#include <unistd.h>
+
+#include "platform-compat.hh"
+#include "logger.hh"
+
+namespace xamarin::android::internal
+{
+	constexpr size_t SENSIBLE_TYPE_NAME_LENGTH = 128;
+	constexpr size_t SENSIBLE_PATH_MAX = 512;
+	// 64-bit unsigned or 64-bit signed with sign
+	constexpr size_t MAX_INTEGER_DIGIT_COUNT_BASE10 = 21;
+
+#if DEBUG
+	constexpr bool BoundsCheck = true;
+#else
+	constexpr bool BoundsCheck = true;
```

Comment:
Why is this value conditional on `#if DEBUG` if both values are the same?  Should this instead be `false`?